odc/stats/tasks.py

```diff
@@ -130,10 +129,23 @@ def __init__(
     def out_path(self, suffix: str) -> str:
         return out_path(suffix, self._output)
 
+    def ds_align(self, dss: Iterable, product: DatasetType, pair_size: int):
+
+        def pair_dss(groups, pair_size):
+            for _, ds_group in groups:
+                ds_group = tuple(ds_group)
+                if len(ds_group) == pair_size:
+                    yield ds_group
+        paired_dss = groupby(dss, key=lambda ds: (ds.center_time, ds.metadata.region_code))
+        paired_dss = pair_dss(paired_dss, pair_size)
+        map_fuse_func = lambda x: fuse_ds(*x, product=product)
+        dss = map(map_fuse_func, paired_dss)
+        return dss
+
```

Comment:
I find this bit confusing.  "Pair" implies 2 of something, but pair_size is a free parameter here.

Also the line:  `paired_dss = pair_dss(paired_dss, pair_size)` seems discard any groups whose size does not match pair_size (rather than trying to partition them down or aggregate them up into subgroups of exact size pair_size).  Is this the intention?

Reply:
I'll change the function name, it's to gather >= 2 datasets in group, hence the free pair_size. And yes that's the intention. If datasets missing from one of products, we won't process that 'group' altogether.

Reply:
done